server/src/WebHook.hs

```diff
@@ -135,6 +149,45 @@ runWebHook uuid logger Config{..} commit =
             defaultLogLevel
         $ do
             postStatus' Pending "Working on checks..."
-            postStatus' Success "No checks implemented yet"
+            handleCompileResult uuid commit config
+            handleTestingResult uuid commit config
+  where
+    postStatus' = postStatus uuid githubToken domain commit
+
+handleCompileResult :: UUID -> Commit -> Config -> LogT IO ()
+handleCompileResult uuid commit Config{..} = do
+    postStatus' Pending "Compilation in progress..."
```

Comment:
I think a single `postStatus` after everything is passed or something went wrong should be enough.